packages/server/src/login/login-course.service.ts

```diff
@@ -31,38 +31,36 @@ export class LoginCourseService {
     const userCourses = [];
 
     for (const c of info.courses) {
-      const course: CourseModel = await this.courseSectionToCourse(
-        c.course,
-        c.section,
-      );
-
-      if (course) {
-        const userCourse = await this.courseToUserCourse(
-          user.id,
-          course.id,
-          Role.STUDENT,
-        );
-        userCourses.push(userCourse);
-      }
-    }
+      if (isKhouryCourse(c)) {
+        const course = await this.courseCRNToCourse(c.crn, c.semester);
 
-    if (info.ta_courses) {
-      for (const c of info.ta_courses) {
-        // Query for all the courses which match the name of the generic course from Khoury
-        const courseMappings = (
-          await CourseSectionMappingModel.find({
-            where: { genericCourseName: c.course }, // TODO: Add semester support
-            relations: ['course'],
-          })
-        ).filter((cm) => cm.course.enabled);
-
-        for (const courseMapping of courseMappings) {
-          const taCourse = await this.courseToUserCourse(
+        if (course) {
+          const userCourse = await this.courseToUserCourse(
             user.id,
-            courseMapping.courseId,
-            c.instructor === 1 ? Role.PROFESSOR : Role.TA,
+            course.id,
+            this.convertKhouryRole(c.role),
           );
-          userCourses.push(taCourse);
+          userCourses.push(userCourse);
+        }
+      } else {
+        // c is a KhouryProfCourse
+        // only need to inspect one of the CRNs from the list
+        // b/c they should all map to the same section group
+        if (c.crns.length !== 0) {
```

Comment:
Can we throw a 400 otherwise? So admin can get an error message if they give us bad data

Reply:
In the interest of getting things moving, I think we can also come back to this when everything else is implemented.